File: frontend-flutter/lib/utils/estudiante.dart
class Estudiante {
  final int? id;
  final String dni;
  final String nombres;
  final String apellidoPaterno;
  final String apellidoMaterno;
  final DateTime fechaNacimiento;
  final String celular;
  final String email;
  final String? direccion;
  final int carreraId;
  final String carreraNombre;
  final String codigoEstudiante;
  final String estado;

  Estudiante({
    this.id,
    required this.dni,
    required this.nombres,
    required this.apellidoPaterno,
    required this.apellidoMaterno,
    required this.fechaNacimiento,
    required this.celular,
    required this.email,
    this.direccion,
    required this.carreraId,
    required this.carreraNombre,
    required this.codigoEstudiante,
    required this.estado,
  });

  factory Estudiante.fromJson(Map<String, dynamic> json) {
    return Estudiante(
      id: json['id'],
      dni: json['dni'] ?? '',
      nombres: json['nombres'] ?? '',
      apellidoPaterno: json['apellido_paterno'] ?? '',
      apellidoMaterno: json['apellido_materno'] ?? '',
      fechaNacimiento: DateTime.parse(json['fecha_nacimiento'] ?? DateTime.now().toIso8601String()),
      celular: json['celular'] ?? '',
      email: json['email'] ?? '',
      direccion: json['direccion'],
      carreraId: json['carrera_id'] ?? 0,
      carreraNombre: json['carrera_nombre'] ?? '',
      codigoEstudiante: json['codigo_estudiante'] ?? '',
      estado: json['estado'] ?? 'activo',
    );
  }

  String get nombreCompleto => '$nombres $apellidoPaterno $apellidoMaterno';
}

File: backend-laravel/app/Models/Estudiante.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Estudiante extends Model
{
    use SoftDeletes;

    protected $table = 'estudiantes';

    protected $fillable = [
        'dni', 
        'nombres', 
        'apellido_paterno', 
        'apellido_materno',
        'fecha_nacimiento', 
        'celular', 
        'email', 
        'direccion',
        'carrera_id', 
        'codigo_estudiante', 
        'estado'
    ];

    public function carrera()
    {
        return $this->belongsTo(Carrera::class);
    }

    public function admisiones()
    {
        return $this->hasMany(Admision::class);
    }

    public function getNombreCompletoAttribute()
    {
        return "{$this->nombres} {$this->apellido_paterno} {$this->apellido_materno}";
    }
}